server/src/main/java/org/opensearch/index/fielddata/IndexFieldData.java

```diff
@@ -150,6 +151,13 @@ public void disableSkipping() {
             this.enableSkipping = false;
         }
 
+        protected Pruning filterPruning(Pruning pruning) {
```

Comment:
Do we actually need to keep `enableSkipping` here? We could just use `Pruning pruning` instead, right?

Reply:
This came up in https://github.com/opensearch-project/OpenSearch/pull/11421#discussion_r1411451813.  However, revisiting this I don't see why we can't replace it with a pruning value of GREATER_THAN by default and change `disableSkipping` to set pruning to None.  @msfroh @gashutos wdyt?
```java
        public void disableSkipping() {
            this.pruning = Pruning.NONE;
        }
```

Reply:
We could also pass in constructor (if that works)

Reply:
Apologies @reta, had to refresh my memory here...

The reason we keep this is so we can mutate the Pruning value based on if `disableSkipping` has been invoked on source before its passed to the actual comparator.  Without that we would pass the Pruning value sent to `newComparator` directly to the actual comparator and ignore the source's config that was introduced with https://github.com/opensearch-project/OpenSearch/pull/8168.  Rather than store a default pruning value that would get ignored, we only need to know if it should be NONE based on if skipping bool is disabled.

Reply:
> We could also pass in constructor (if that works)

We could but I think we would still need to ignore the default on the source if its either `GREATER_THAN` or `GREATER_THAN_OR_EQUAL_TO` to honor what is passed to newComparator.

Reply:
Got it, thanks @mch2 , I think we could address it later on if needed

Reply:
IIRC, the idea was that we are operating at two different levels:
1. The Lucene level where we now have 3 possible values for `Pruning` (where one value is "off", and the other two are forms of "on").
2. The OpenSearch level where we enable or disable skipping (i.e it's just "off" or "on"), where an "off" value forces the same behavior as `Pruning.NONE` and the "on" value leaves the `Pruning` value as-is.

While we could arbitrarily pick one of the "on" `Pruning` values to get the same OpenSearch-level behavior, I liked the idea of keeping it as a `boolean` to reflect the fact that we only have two possible states at the OpenSearch level.